android/src/main/java/com/christopherdro/htmltopdf/RNHTMLtoPDFModule.java

```diff
@@ -32,57 +42,59 @@ public String getName() {
   public void convert(final ReadableMap options, final Promise promise) {
     try {
       File destinationFile;
-      String htmlString = options.hasKey("html") ? options.getString("html") : null;
-      if (htmlString == null) return;
+      String htmlString = options.hasKey(HTML) ? options.getString(HTML) : null;
+      if (htmlString == null) {
+        promise.reject(new Exception("RNHTMLtoPDF error: Invalid htmlString parameter."));
+        return;
+      }
 
       String fileName;
-      if (options.hasKey("fileName")) {
-        fileName = options.getString("fileName");
+      if (options.hasKey(FILE_NAME)) {
+        fileName = options.getString(FILE_NAME);
+        if (!isFileNameValid(fileName)) {
+          promise.reject(new Exception("RNHTMLtoPDF error: Invalid fileName parameter."));
+          return;
+        }
       } else {
         fileName = UUID.randomUUID().toString();
       }
 
-      if (options.hasKey("directory") && options.getString("directory").equals("docs")) {
+      if (options.hasKey(DIRECTORY)) {
         String state = Environment.getExternalStorageState();
-          File path = (Environment.MEDIA_MOUNTED.equals(state)) ?
-                  new File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_DOCUMENTS)
-                  : new File(mReactContext.getFilesDir(), Environment.DIRECTORY_DOCUMENTS);
+        File path = (Environment.MEDIA_MOUNTED.equals(state)) ?
+          new File(Environment.getExternalStorageDirectory(), options.getString(DIRECTORY)) :
+          new File(mReactContext.getFilesDir(), options.getString(DIRECTORY));
 
         if (!path.exists()) path.mkdir();
-        destinationFile = new File(path, fileName + ".pdf");
+        destinationFile = new File(path, fileName + PDF_EXTENSION);
       } else {
-        destinationFile = getTempFile(fileName);
+        destinationFile = getTempFile();
       }
 
       convertToPDF(htmlString,
               destinationFile,
-              options.hasKey("base64") && options.getBoolean("base64") == true,
+              options.hasKey(BASE_64) && options.getBoolean(BASE_64),
               Arguments.createMap(),
               promise,
-              options.hasKey("baseURL") ? options.getString("baseURL") : null);
+              options.hasKey(BASE_URL) ? options.getString(BASE_URL) : null);
     } catch (Exception e) {
-      promise.reject(e.getMessage());
+      promise.reject(e);
```

Comment:
Any reason to reject with the error object vs. the message?

Returning the error message itself keeps things inline with the iOS implementations.
https://github.com/christopherdro/react-native-html-to-pdf/blob/master/ios/RNHTMLtoPDF/RNHTMLtoPDF.m#L153